src/shredder.rs

```diff
@@ -525,8 +558,9 @@ fn create_output_shreds_for_other_leader(
         shreds.push(ValidatedShred::new_validated(shred));
     }
     for mut c in coding {
-        c.0.index_in_slice += num_data_shreds;
-        let merkle_path = tree.create_proof(c.0.index_in_slice);
+        // based on assertion above, this unwrap() should be safe
```

Comment:
in addition to the assertion above (comparing to `TOTAL_SHREDS`) we implicitly assume that the `shred_index`es of the `CodingShred`s are `0..coding.len()`

Reply:
one way to possibly completely circumvent this entire problem, would be to just use the indices within `data` and `coding` instead, i.e. the shreds in `data` get `ShredIndex`es `0..data.len()` and the ones in `coding` get `data.len()..TOTAL_SHREDS`?!